### PR TITLE
Changes for AutoGluon Doc

### PR DESCRIPTION
**Pull request by zhanghang1989**
AutoGluon Doc can be previewed at http://autogluon.s3.amazonaws.com/index.html#

There are main to changes:

- Enables 'class', 'func', 'meth' keywords for rst
- Add `autoautosummary` method:

```
.. autoclass:: FIFOScheduler
   :members:
   :inherited-members:

    .. rubric:: Methods

    .. autoautosummary:: FIFOScheduler
        :methods:
```

The changes should not break original d2lbook.

### CODE REVIEW

File: d2lbook/sphinx_template.py [hunk 1]
```diff
@@ -1,6 +1,13 @@
 
 sphinx_conf = r"""
 import sys
+from sphinx.ext.autosummary import Autosummary
+from sphinx.ext.autosummary import get_documenter
+from docutils.parsers.rst import directives
+from sphinx.util.inspect import safe_getattr
```
Comment (mli) [inline d2lbook/sphinx_template.py]: do these imports need to install additional packages?

File: d2lbook/sphinx_template.py [hunk 2]
```diff
@@ -84,13 +91,53 @@
 }
 
 SPHINX_CONFIGS
+class AutoAutoSummary(Autosummary):
```
[the remark this answers is not in the document]
Comment (zhanghang1989) [reply d2lbook/sphinx_template.py]: It automatically generates summary of the class and its methods and attributes without  calling sphinx-autogen. One example:
http://autogluon.s3.amazonaws.com/api/autogluon.scheduler.html#autogluon.scheduler.FIFOScheduler

File: d2lbook/sphinx_template.py (hunk 1 above)
Comment (zhanghang1989) [reply d2lbook/sphinx_template.py]: No additional packages.  sphinx built-in

File: d2lbook/sphinx.py [hunk 3]
```diff
@@ -24,7 +24,8 @@ def prepare_env(self):
         self._update_pyconf('index', self.config.build['index'])
         self._update_pyconf('sphinx_configs', self.config.build['sphinx_configs'])
         extensions = ['recommonmark', 'sphinxcontrib.bibtex',
-                      'sphinxcontrib.rsvgconverter', 'sphinx.ext.autodoc']
+                      'sphinxcontrib.rsvgconverter', 'sphinx.ext.autodoc',
```
[the remark this answers is not in the document]
Comment (zhanghang1989) [reply d2lbook/sphinx.py]: Yeah, I will remove that.

**Comment by zhanghang1989** [comment]
close this PR for now, due to other changes in autogluon doc

**Pull request closed without merging**